convert_hf_to_gguf.py

```diff
@@ -7928,6 +7928,62 @@ def prepare_tensors(self):
             if len(experts) > 0:
                 raise ValueError(f"Unprocessed experts: {experts}")
 
+
+@ModelBase.register("CogVLMForCausalLM")
+class CogVLMVisionModel(MmprojModel):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hparams['num_attention_heads'] = self.hparams['num_heads']
+
+    def set_gguf_parameters(self):
+        super().set_gguf_parameters()
+        self.gguf_writer.add_vision_attention_layernorm_eps(self.hparams.get("layer_norm_eps", 1e-6))
+        self.gguf_writer.add_clip_projector_type(gguf.VisionProjectorType.COGVLM)
+
+    def modify_tensors(self, data_torch: Tensor, name: str, bid: int | None) -> Iterable[tuple[str, Tensor]]:
+        del bid  # unused
+
+        if not name.startswith("model.vision."):
+            return []
+
+        if "query_key_value" in name:
+            # Split tensor into three along first axis
+            q, k, v = data_torch.split(data_torch.shape[0] // 3, dim=0)
+            return [
+                (self.map_tensor_name(name.replace("query_key_value", "query")), q),
+                (self.map_tensor_name(name.replace("query_key_value", "key")), k),
+                (self.map_tensor_name(name.replace("query_key_value", "value")), v),
+            ]
+
+        return [(self.map_tensor_name(name), data_torch)]
```

Comment:
```suggestion
        return [(self.map_tensor_name(name), data_torch)]
```
Create Q/K/V views at build time instead (check other (non-mm) models for examples).